app/notebook/server/NotebookManager.scala

```diff
@@ -75,9 +77,11 @@ class NotebookManager(val name: String, val notebookDir: File) {
     nbData.map { nb =>
       val newPath = incrementFileName(nb._4.dropRight(extension.length))
       val newName = getName(newPath)
-      NBSerializer.read(nb._3) match {
+      Notebook.deserialize(nb._3) match {
```

Comment:
what's `nb._3` ? Maybe we can use pattern match on the `map` to extract the members using readable names?